part2/bowling/frame.rb

```diff
@@ -0,0 +1,40 @@
+class Frame
+  attr_accessor :goes, :extra
+
+  def initialize(pins = nil)
+    @goes = []
+    @extra = []
+
+    if pins
+      goes << pins
+    end
+  end
+
+  def original_score
+    goes.inject(0, :+)
+  end
+
+  def extra_score
+    extra.inject(0, :+)
+  end
+
+  def spare?
+    goes.length >= 2 && goes[0] + goes[1] == 10
+  end
+
+  def strike?
+    goes.length >= 1 && goes[0] == 10
+  end
+
+  def <<(pins)
+    goes << pins
+  end
```

Comment:
Nice idea, but I feel worried about giving `<<` new functionality on top of its normal functionality, because it seems easy to forget (a) that you did this and (b) whether any given instance of `<<` is the new one or the normal one.